### PR TITLE
python310Packages.fvcore: init at 0.1.5

### PR DESCRIPTION
**Pull request by fennecs-system**
###### Description of changes

fvcore is a library that provides common functionality shared in various computer vision frameworks developed by Facebook AI Research. Additionally iopath is a dependency of fvcore. This pull request adds these projects into nixpkgs.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by kilianar** [comment]
There are several trailing white space issues. You should remove them. The title of your commit should indicate the version. For more information, refer to the following link: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md.

File: pkgs/development/python-modules/fvcore/default.nix [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+    "tests"
+  ];
+
+  pythonImportsCheck = [
+    "fvcore"
+    "fvcore.nn"
+  ];
+
+  meta = with lib; {
+    description = "Collection of common code that's shared among different research projects in FAIR computer vision team.";
```
Comment (kilianar) [inline pkgs/development/python-modules/fvcore/default.nix:63]: Don’t include a period at the end. (See https://nixos.org/manual/nixpkgs/unstable/#var-meta-description)

File: pkgs/development/python-modules/iopath/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  # tests depend on other packages like torch
+  doCheck = false;
+
+  pythonImportsCheck = [
+    "iopath"
+  ];
+
+  meta = with lib; {
+    description = " A python library that provides common I/O interface across different storage backends. ";
```
Comment (kilianar) [inline pkgs/development/python-modules/iopath/default.nix:38]: Don’t include a period at the end. (See https://nixos.org/manual/nixpkgs/unstable/#var-meta-description)

You should also remove the leading and trailing white space in description.

File: pkgs/development/python-modules/fvcore/default.nix [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    homepage = "https://github.com/facebookresearch/fvcore";
+    license = licenses.asl20;
+    maintainers = with maintainers; [ swaddle ];
```
Comment (kilianar) [inline pkgs/development/python-modules/fvcore/default.nix:66]: Please add yourself to the list of maintainers in the file `maintainers/maintainer-list.nix`. There are informative comments on how to do so at the start of the file.

File: pkgs/development/python-modules/fvcore/default.nix [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+buildPythonPackage rec {
+  pname = "fvcore";
+  disabled = pythonOlder "3.6";
+  version = "0.1.5";
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "fvcore";
+    rev = version;
```
Comment (kilianar) [inline pkgs/development/python-modules/fvcore/default.nix:29]: It seems like there isn't a tag or release with the specified version number. When attempting to build the package, the process fails with a 404 error. Here is the error message:
```
trying https://github.com/facebookresearch/fvcore/archive/0.1.5.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
^M  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0^M  0    14    0     0    0     0      0      0 --:--:--  0:00:01 --:--:--     0
curl: (22) The requested URL returned error: 404
error: cannot download source from any mirror
```

**Comment by kilianar** [comment]
It looks like many of the dependencies in your package definition aren't being utilized. Just double-check if they are necessary and need to be included in `{native,propagated,}buildInputs`, or if they can be removed entirely.

File: pkgs/development/python-modules/fvcore/default.nix [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+buildPythonPackage rec {
+  pname = "fvcore";
+  disabled = pythonOlder "3.6";
+  version = "0.1.5";
+
+  src = fetchFromGitHub {
+    owner = "facebookresearch";
+    repo = "fvcore";
+    rev = "6bde555";
+    sha256 = "1vz9qif5iaxhwks5mhg9lrfr16aj5kb108dgphhcv0iq4nla2wsa";
```
Comment (pbsds) [inline pkgs/development/python-modules/fvcore/default.nix:29]: Please use the new hash format: `hash = "sha256-something";`

File: pkgs/development/python-modules/fvcore/default.nix [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  nativeBuildInputs = [
```
Comment (pbsds) [inline pkgs/development/python-modules/fvcore/default.nix:32]:
```suggestion
  propagatedBuildInputs = [
```

File: pkgs/development/python-modules/fvcore/default.nix [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+    pyyaml
+    tabulate
+    termcolor
+    tqdm
+    yacs
+  ];
+
+  propagatedBuildInputs = [
+    torch
+    shapely
```
Comment (pbsds) [inline pkgs/development/python-modules/fvcore/default.nix:45]:
```suggestion
```

shapely seems to be an optional dependency.

Please add it to `passthrou` ([how](https://nixos.org/manual/nixpkgs/stable/#python-optional-dependencies)), then maybe ++ it to the checkinputs if needed.

File: pkgs/development/python-modules/fvcore/default.nix [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+    pillow
+    pyyaml
+    tabulate
+    termcolor
+    tqdm
+    yacs
+  ];
+
+  propagatedBuildInputs = [
+    torch
```
Comment (pbsds) [inline pkgs/development/python-modules/fvcore/default.nix:44]:
```suggestion
```

[They consider it a "dependency"](https://github.com/facebookresearch/fvcore/blob/0f2b23b6f93e36041d9a74764ee824541cf0a0e5/.circleci/config.yml#L44), but it is only needed [in `fcvore.nn` and in tests](https://github.com/search?q=repo%3Afacebookresearch%2Ffvcore+%2Fimport+torch%2F&type=code). 

Consider adding it as an optional dependency, as with shapely.

File: pkgs/development/python-modules/fvcore/default.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/development/python-modules/fvcore/default.nix:62]:
```suggestion
    description = "Light-weight core library used in FAIR computer vision research projects";
```

File: pkgs/development/python-modules/iopath/default.nix (hunk 2 above)
Comment (pbsds) [reply pkgs/development/python-modules/iopath/default.nix:36]:
```suggestion
    description = "Common I/O interface across various storage backends";
```

**Review by pbsds — COMMENTED** [review]
Great work, and welcome to nixpkgs!

Please rebase and squash your commits into three commits titled as follows:

* `maintainers: add swaddle`
* `python3Packages.iopath: init at 0.1.9`
* `python3Packages.fvcore: init at 0.1.5`

You also need to rebase your changes on top of master, due to merge conflicts

**Pull request merged**